stdlib/public/runtime/MetadataLookup.cpp

```diff
@@ -425,19 +426,20 @@ _findProtocolDescriptor(llvm::StringRef mangledName) {
 
 #pragma mark Type field descriptor cache
 namespace {
-template <typename T> struct FieldDescriptorCacheEntry {
+struct FieldDescriptorCacheEntry {
 private:
-  const Metadata *Base;
-  const T *Description;
+  const Metadata *Type;
+  const FieldDescriptor *Description;
 
 public:
-  FieldDescriptorCacheEntry(const Metadata *Base, const T *description)
-  : Base(Base), Description(description) {}
+  FieldDescriptorCacheEntry(const Metadata *type,
+                            const FieldDescriptor *description)
+      : Type(type), Description(description) {}
 
-  const T *getDescription() { return Description; }
+  const FieldDescriptor *getDescription() { return Description; }
 
   int compareWithKey(const Metadata *other) const {
-    return Base == other;
+    return Type == other ? 0 : (Type < other ? -1 : 1);
```

Comment:
Comparing unrelated pointers is UB in C/C++. Strictly, you need to do `(uintptr_t)Type < (uintptr_t)Other` there.

Reply:
Ah interesting, thanks!

Reply:
Fixed!